src/auto-launch.js

```diff
@@ -1,29 +1,86 @@
 import AutoLaunch from 'auto-launch'
+import { app } from 'electron'
+import os from 'os'
+import path from 'path'
+import fs from 'fs-extra'
+import untildify from 'untildify'
 import createToggler from './create-toggler'
 import logger from './common/logger'
 import store from './common/store'
 
 const CONFIG_KEY = 'autoLaunch'
 
-const autoLauncher = new AutoLaunch({
-  name: 'IPFS Desktop'
-})
+function isSupported () {
+  const plat = os.platform()
+  return plat === 'linux' || plat === 'win32' || plat === 'darwin'
+}
+
+// Disable the old auto launch mechanism.
+// TODO: remove on 0.10.0.
+async function disableOldLogin () {
+  try {
+    const autoLauncher = new AutoLaunch({ name: 'IPFS Desktop' })
+
+    if (await autoLauncher.isEnabled()) {
+      await autoLauncher.disable()
+      logger.error('[launch on startup] old mechanism disabled')
+    }
+  } catch (_) {
+    // ignore...
+  }
+}
+
+function getDesktopFile () {
+  return path.join(untildify('~/.config/autostart/'), 'ipfs-desktop.desktop')
+}
+
+async function enable () {
+  if (app.setLoginItemSettings) {
+    app.setLoginItemSettings({ openAtLogin: true })
+  }
+
+  const desktop = `[Desktop Entry]
+Type=Application
+Version=1.0
+Name=IPFS Desktop
+Comment=IPFS Desktop Startup Script
+Exec="${process.execPath}"
+StartupNotify=false
+Terminal=false`
+
+  await fs.outputFile(getDesktopFile(), desktop)
+}
+
+async function disable () {
+  if (app.setLoginItemSettings) {
+    app.setLoginItemSettings({ openAtLogin: false })
+  }
+
+  await fs.remove(getDesktopFile())
+}
+
+export default async function (ctx) {
+  await disableOldLogin()
 
-export default function (ctx) {
   const activate = async (value, oldValue) => {
     if (process.env.NODE_ENV === 'development') {
       logger.info('[launch on startup] unavailable during development')
       return
     }
 
+    if (!isSupported()) {
+      logger.info('[launch on startup] not supported on this platform')
+      return false
+    }
```

Comment:
Are we hiding the "Launch on Startup" Option from WebUI's Settings when platform is not supported?

Reply:
Not sure, but will check.